slf4j-tinylog/src/main/java/org/tinylog/slf4j/TinylogSlf4jServiceProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Martin Winandy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.tinylog.slf4j;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.IMarkerFactory;
+import org.slf4j.helpers.BasicMarkerFactory;
+import org.slf4j.spi.MDCAdapter;
+import org.slf4j.spi.SLF4JServiceProvider;
+
+/**
+ * SPI implementation for SLF4j 1.8+. Replacement mechanism for {@link org.slf4j.impl.StaticLoggerBinder}.
+ */
+public class TinylogSlf4jServiceProvider implements SLF4JServiceProvider {
+
+	/**
+	 * SLF4J API 1.8 and newer is supported.
+	 */
+	public static final String REQUESTED_API_VERSION = "1.8";
```

Comment:
Maybe we have to replace the API version to 2.0, if SLF4J 1.8 and 2.0 are not compatible.

Reply:
At this stage, it does not look like SLF4J has already defined compatible versions, or updated the compatibility list:

```
    /**
     * It is LoggerFactory's responsibility to track version changes and manage
     * the compatibility list.
     * <p>
     * <p>
     * It is assumed that all versions in the 1.6 are mutually compatible.
     */
    static private final String[] API_COMPATIBILITY_LIST = new String[] { "1.8", "1.7" };
```

https://github.com/qos-ch/slf4j/blob/e2c29c81df6fb405a3c5a6dfd1cf28ba26a7c1d6/slf4j-api/src/main/java/org/slf4j/LoggerFactory.java#L110-L117

So in theory we could also report 1.7, which is pretty confusing, because that would mean `TinylogSlf4jServiceProvider` requests an API level in which it could not have existed, because `SLF4JServiceProvider` did not exist. That why I believe 1.8 is the correct number to report right now.